### ISSUE TITLE
Js plugins ane not working properly in prod build in angular

### ISSUE THREAD

**Issue by bhartendukumar12** [videojs/video.js#5384]
I have imported videojs module in angular application (V4).The video is playing perfectly in 360 mode when i give a normal build.The moment I give prod build the video does not play in 360.

**Comment by Akxe** [comment]
You gave 0 information, try to fill the issue template. Also, this isn't problem of video.js, but of Angular's webpack, so ask there.

**Comment by misteroneill** [comment]
There could be an issue with the way Video.js' modules are consumed by Angular, but without more information and a reduced test case, we cannot guess what the issue might be.

**Comment by Akxe** [comment]
Also to be honest, having videojs alongside with angular is kinda bad, as a lot of code in videojs is duplicated by therefor increasing your time till interactive.

**Comment by gkatsev** [comment]
We've had issues with minifcation previously due to typeofs and uglify 2, which could be the issue since that's a thing in prod builds but yeah, without more info, we can't do more.
@Akxe what do you mean that videojs along side angular is bad?

**Comment by Akxe** [comment]
First of all you lose a lot of type safety when using videos, as the library wasn’t build with that in mind, but ok... it ain’t bad that since official `.d.ts` file. 
My point was, that when you want to use videojs you’ll pay cost, the library is quite big. A lot is somewhat duplicate with what angular has, therefor if there would be some angular version of video library, you wouldn’t have to pay the overhead twice. 
And finally integrating libraries in angular seems kinda hell to me... but it might have changed since the last time.

To summarize, my biggest concern is functionality redundancy, and lost of type safety.

**Comment by gkatsev** [comment]
Yeah, Video.js isn't small but it provides a lot of functionality that you want in a video player. Also, I'm not sure that there's *that* much shared code that's redundant. The main biggest parts of the code are HLS/DASH support, text track support, and controls, probably in that order.
Also, I'm not sold that type safety is that important here, since Video.js is fairly self contained :)
Thanks for elaborating, though!

**Comment by Poseclop** [comment]
Not sure if this is related but I'm also having issues with videoJS in angular prod build (angular 9+).
Preload attribute works fine in dev mode. It also works fine in simulated prod (ng serve --prod).
However, once project is build in prod (ng build --prod), preload is no longer working. Chrome download the full video before starting to play it and download it again if I navigate in video. Just to confirm, all test are done with the same media server and response headers seem ok.
I tried to update my code to match the demo on this Github but no change.
I also tried to pass all properties to the player using the player instance instead of template or options but again, same thing.
